lib/src/render_picture.dart

```diff
@@ -161,41 +165,84 @@ class RenderPicture extends RenderBox {
     if (picture == null || size == Size.zero) {
       return;
     }
-    context.canvas.save();
-    context.canvas.translate(offset.dx, offset.dy);
-    if (_flipHorizontally) {
-      context.canvas.translate(size.width, 0.0);
-      context.canvas.scale(-1.0, 1.0);
-    }
-
-    // this is sometimes useful for debugging, e.g. to draw
-    // a thin red border around the drawing.
-    assert(() {
-      if (RenderPicture.debugRectColor != null &&
-          RenderPicture.debugRectColor!.alpha > 0) {
-        context.canvas.drawRect(
-            Offset.zero & size,
-            Paint()
-              ..color = debugRectColor!
-              ..style = PaintingStyle.stroke);
-      }
-      return true;
-    }());
-    scaleCanvasToViewBox(
-      context.canvas,
-      size,
+    context.pushLayer(OffsetLayer(offset: offset), (context, offset) {
+      var transform = scaleMatrixToViewBox(
+        size,
       _picture!.viewport,
       _picture!.size,
+      );
+      if (transform == null) {
+        context.addLayer(PictureLayerNoDispose(Rect.fromLTWH(offset.dx, offset.dy, size.width, size.height))
+          ..isComplexHint = true
+          ..picture = picture!.picture
+        );
+      } else {
+        context.pushTransform(needsCompositing, offset, transform, (context, offset) {
+          context.addLayer(PictureLayerNoDispose(Rect.fromLTWH(offset.dx, offset.dy, size.width, size.height))
+            ..isComplexHint = true
+            ..picture = picture!.picture
+          );
+        });
+      }
+
+    }, offset);
+
+
+    // scaleCanvasToViewBox(
+    //   context.canvas,
+    //   size,
+    //   _picture!.viewport,
+    //   _picture!.size,
+    // );
+    // final Rect viewportRect = Offset.zero & _picture!.viewport.size;
+    // if (allowDrawingOutsideViewBox != true) {
+    //   context.canvas.clipRect(viewportRect);
+    // }
+    // context.addLayer(PictureLayer(Rect.fromLTWH(offset.dx, offset.dy, size.width, size.height))
+    //   ..isComplexHint = true
+    //   ..picture = picture!.picture
+    // );
+    // context.canvas.restore();
+  }
+}
+
+class PictureLayerNoDispose extends PictureLayer {
```

Comment:
Can you avoid this hack by keeping a reference to the PictureLayer in the RenderObject via a LayerHandle: https://master-api.flutter.dev/flutter/rendering/LayerHandle-class.html.

It should avoid that the layer and its resources get disposed, which is presumably causing the segfault you were seeing?

(or store the layer in RenderObject.layer, if it is not already used)